### PR TITLE
Add exception for unique_id to ADR0007

### PR DESCRIPTION
**Pull request by ludeeus**
Ref: <https://github.com/home-assistant/core/pull/58596#issuecomment-1004117319>

### CODE REVIEW

**Review by frenck — APPROVED** [review]

File: adr/0007-integration-config-yaml-structure.md [hunk 1]
```diff
@@ -47,6 +47,10 @@ sensor:
     username: user
 ```
 
+### Exceptions
+
+- Implementing an optional `unique_id` configuration key to set the unique ID of the entity created by the platform is allowed.
```
Comment (MartinHjelmare) [inline adr/0007-integration-config-yaml-structure.md:52]: Should we say something about what platforms are allowed to do this. It's not allowed for all platforms, only platforms that create entities that derive their state from other integrations or platforms for transport implementing integrations, or if we have made some special exception.

Comment (frenck) [reply adr/0007-integration-config-yaml-structure.md:52]: Maybe say: May not conflict with ADR-0010?

Comment (ludeeus) [reply adr/0007-integration-config-yaml-structure.md:52]: It has been added to `command_line` that is not excepted from ADR0010 or derived from other entities or is a transport protocol, so changing the wording here does not reflect what has already been accepted.

Comment (frenck) [reply adr/0007-integration-config-yaml-structure.md:52]: The command line defines its own "protocol". As in, it does not connect to a device or service in a normal sense. It is safe to say it doesn't fall under adr-0010 (at least not intentional)

Comment (ludeeus) [inline adr/0007-integration-config-yaml-structure.md:52]:
```suggestion
- For integrations that do not fall under [ADR-0010](https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md), implementing an optional `unique_id` configuration key to set the unique ID of the entity created by the platform is allowed.
```

**Review by MartinHjelmare — APPROVED** [review]

**Review by frenck — APPROVED** [review]

**Pull request merged**